lib/saluki-components/src/transforms/host_tags/mod.rs

```diff
@@ -124,3 +129,44 @@ impl SynchronousTransform for HostTagsEnrichment {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use std::{
+        num::NonZeroUsize,
+        time::{Duration, Instant},
+    };
+
+    use bytesize::ByteSize;
+    use saluki_context::{Context, ContextResolverBuilder};
+    use saluki_event::metric::Metric;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_host_tags_enrichment() {
+        let context_resolver = ContextResolverBuilder::from_name("host_tags")
+            .expect("resolver name is not empty")
+            .with_interner_capacity_bytes(NonZeroUsize::new(ByteSize::kib(64).as_u64() as usize).unwrap())
+            .build();
+
+        let mut host_tags_enrichment = HostTagsEnrichment {
+            start: Instant::now(),
+            context_resolver: Some(context_resolver),
+            expected_tags_duration: Duration::from_secs(30),
+            host_tags: Some(vec!["hosttag1".to_string(), "hosttag2".to_string()]),
```

Comment:
It would be nice to lift these out from their inline definition, so we can then actually check that the enriched metric has these as the tags.